resourcepolicies.md

```diff
@@ -10,34 +10,298 @@ As we don't have yet an use case to iterate over all the existent resource polic
+
+the add operation will result in:
+
+```json
+  "id": 2844,
+  "name": "my name",
+  "description": "my description",
+  "policyType": "TYPE_SUBMISSION",
+  "action": "READ",
+  "startDate" : "2019-10-31",
+  "endDate" : null,
+  "type": "resourcepolicy"
+```
+
+### Remove
+With the `remove` operation, you can delete:
+- startDate
+- endDate
+- name
+- description
+
+the other properties cannot be nullified.
+
+To remove an embargo you can for instance use the following
+`curl --data '[ { "op": "remove", "path": "/startDate" }]' -H "Authorization: Bearer ..." -H "content-type: application/json" -X PATCH ${dspace7-url}/api/authz/resourcepolicies/${resourcepolicy-id}`
+
+that will transform
+
+```json
+  "id": 2844,
+  "name": "my name",
+  "description": "my description",
+  "policyType": "TYPE_SUBMISSION",
+  "action": "READ",
+  "startDate" : "2019-10-31",
+  "endDate" : null,
+  "type": "resourcepolicy"
+```
+
+in 
+
+```json
+  "id": 2844,
+  "name": "my name",
+  "description": "my description",
+  "policyType": "TYPE_SUBMISSION",
+  "action": "READ",
+  "startDate" : null,
+  "endDate" : null,
+  "type": "resourcepolicy"
+```
+
+Use of the remove operation over a not acceptable path will result in a 422 Unprocessable Entity return code
+
+### Replace
+The replace operation allows to replace *existent* information with new one. Attempt to use the replace operation to set not yet initialized information must return an error. See [general errors on PATCH requests](patch.md)
+
+To change the startDate
+`curl --data '[ { "op": "replace", "path": "/startDate", "value": "2020-01-01" }]' -H "Authorization: Bearer ..." -H "content-type: application/json" -X PATCH ${dspace7-url}/api/authz/resourcepolicies/${resourcepolicy-id}`
+
+For example, starting with the following item data:
+
+```json
+  "id": 2844,
+  "name": null,
+  "description": null,
+  "policyType": "TYPE_SUBMISSION",
+  "action": "READ",
+  "startDate" : "2019-10-31",
+  "endDate" : null,
+  "type": "resourcepolicy"
+```
+
+the replace operation will result in:
+
+```json
+  "id": 2844,
+  "name": null,
+  "description": null,
+  "policyType": "TYPE_SUBMISSION",
+  "action": "READ",
+  "startDate" : "2020-01-01",
+  "endDate" : null,
+  "type": "resourcepolicy"
+```
+
+## Deleting a resource policy
+**DELETE /api/authz/resourcepolicies/<:id>**
+
+Return codes:
+* 204 No content - if the operation succeed
+* 401 Forbidden - if you are not authenticated
+* 403 Unauthorized - if you are not logged in with sufficient permissions. Only system administrators or user with ADMIN permission over the target resource can use the endpoint
+* 404 Not found - if the item doesn't exist (or was already deleted)
+
+
 ## Linked entities
 ### EPerson
 **/api/authz/resourcepolicies/<:id>/eperson**
 
-t.b.d
+Provide access to the eperson linked by this resource policy
+
+The recipient of the policy, eperson or group, cannot be modified. If you need to do so please delete the policy and create a new one.
```

Comment:
Can you add permission requirements and return codes (especially since it's nullable) for this
If regular users would be authorized to retrieve resource policies, can they also access all data about this object?

Reply:
Can you add the response code when requesting /api/authz/resourcepolicies/<:id>/eperson for a resource policy which doesn't contain an eperson as well?
It will also be important to ensure these objects are not embedded

The same applies for the 2 comments below

Reply:
done

Reply:
About the embedding I would prefer to don't say anything special here for the following reasons:
1) we usually don't include this information in the contract. Embedding is a standard HAL format capability and ideally we should be able to include/exclude embedding via projection also "on demand"
2) if you worry about how much information ("all data") are exposed about the linked object this is not something that need to be solved (and stated) in the single endpoint as we should provide an uniform behavior via an authorization interceptor that will "clean" the response hiding data that are not allowed to the user
3) if you worry about infinite loop, please check this proposal https://github.com/DSpace/Rest7Contract/issues/2#issuecomment-528386269 as I guess is the way to go. In any case we can postpone the exposure of the eperson or other linked object in the implementation until we solve the issue but it makes senso to be uniform in the contract and expect to always have such objects embedded when available and needed (projection)

Reply:
Another reason to be careful about embedding is the fact that so far no permissions are verified for embedded information. This is a problem which should still be resolved but has an even higher risk for this functionality.
If the implementation doesn't embed the linked data, this will minimize the risks on any of these potential issues mentioned here

Reply:
@benbosman as discussed in the webcall, it is the #2 in my "reasons list". I have created the following ticket https://jira.lyrasis.org/browse/DS-4394 to remember about it

Reply:
@benbosman I just wanted to pop in here and note that, via link repositories, `PreAuthorize` annotations *do* work now for embedded subresources. I am not sure what changed, but I tested and observed this to work in this PR: https://github.com/DSpace/DSpace/pull/2625